common/djangoapps/third_party_auth/decorators.py

```diff
@@ -30,3 +34,52 @@ def wrapped_view(request, *args, **kwargs):
         resp['X-Frame-Options'] = x_frame_option
         return resp
     return wraps(view_func, assigned=available_attrs(view_func))(wrapped_view)
+
+
+def tpa_hint_ends_existing_session(func):
+    """
+    Modifies a view function so that, if a tpa_hint URL parameter is received, the user is
+    already logged in, and the hinted SSO provider is so configured, the user is redirected
+    to a logout view and then back here, so that we can ensure that they're using the most
```

Comment:
Can you just call the logout API method, `django.contrib.auth.logout(request)` (as suggested by the django docs) rather than redirecting users and doing this complex URL-building? Then you could also ditch the extra `logged_out` parameter, and the code would be simpler.